doc/aaa/TACACS+ Requirement.md

```diff
@@ -0,0 +1,161 @@
+# SONiC TACACS+ Protocol Requirement
+
+# Table of Contents
+- [Table of Contents](#table-of-contents)
+- [About this Manual](#about-this-manual)
+- [1 Functional Requirements](#1-functional-requirement)
+  * [1.1 Authentication](#11-authentication)
+  * [1.2 Authorization](#12-authorization)
+  * [1.3 Accounting](#13-accounting)
+  * [1.4 User script](#14-user-script)
+  * [1.5 Docker support](#15-docker-support)
+  * [1.6 Multiple TACAS server](#16-multiple-tacas-server)
+- [2 Configuration and Management Requirements](#2-configuration-and-management-requirements)
+  * [2.1 SONiC CLI](#21-sonic-cli)
+  * [2.2 Config DB](#22-config-db)
+  * [2.3 Counter](#23-counter)
+  * [2.4 System log](#24-system-log)
+- [3 Limitation](#limitation)
+  * [3.1 Command size](#31-command-size)
+  * [3.2 Server count](#32-server-count)
+
+# About this Manual
+This document provides a detailed description on the requirement of TACACS+ protocol support.
+
+# 1 Functional Requirement
+## 1.1 Authentication
+- Authentication when user login to SONiC host.
+
+
+## 1.2 Authorization
+- Authorization when:
+	- User login to SONiC host.
+	- User run command on SONiC host.
+		- User can only run commands in whitelist.
+		- For recursive command/script, only the top level command have authorization.
+
+- Only command in whitelist  visible to user:
+	- All privilege level share same whitelist to simplify design.
+	- Different privilege level have different permission to run these command.
+	- All commands in sudoers will add to the whitelist. and sudoers config file still need for RO users, this is because when remote TACACS server not avaliable, we need use local group permission for failover.
+
+- Disable user behavior in shell:
```

Comment:
why rbash is a must for command authorization?

Reply:
The reason is because for command authorization, we create a symbol link for every command we want enable authorization. 
Without rbash user can run any command, include those command we not create symbol link. 
Then they can bypass the authorization very easy.